internal/worker.go

```diff
@@ -159,6 +159,17 @@ type (
 		// Optional: If set defines maximum amount of time that workflow task will be allowed to run. Defaults to 1 sec.
 		DeadlockDetectionTimeout time.Duration
 
+		// Optional: The maximum amount of time between sending each pending heartbeat to the server. Regardless of
+		// heartbeat timeout, no pending heartbeat will wait longer than this amount of time to send.
+		// default: 60 seconds
+		MaxHeartbeatThrottleInterval time.Duration
+
+		// Optional: The default amount of time between sending each pending heartbeat to the server. This is used if the
+		// ActivityOptions do not provide a HeartbeatTimeout. Otherwise, the interval becomes a value a bit smaller than the
+		// given HeartbeatTimeout.
+		// default: 30 seconds
+		DefaultHeartbeatThrottleInterval time.Duration
```

Comment:
Maybe mention that if heartbeatTimeout is provided than the throttle interval is 80% of that timeout?

Reply:
> Otherwise, the interval becomes a value a bit smaller than the given HeartbeatTimeout.

To clarify, are you wanting me to change "a bit smaller than" to "80% of"? I intentionally left off the exact amount as an implementation detail, but can add it if I must.

Reply:
Scratch that, we don't have to document the exact value.